src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityContext.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// ActivityContext representation conforms to the w3c TraceContext specification. It contains two identifiers
+    /// a TraceId and a SpanId - along with a set of common TraceFlags and system-specific TraceState values.
+    /// </summary>
+    public readonly struct ActivityContext : IEquatable<ActivityContext>
+    {
+        /// <summary>
+        /// Construct a new object of ActivityContext.
+        /// </summary>
+        /// <param name="traceId"> A trace identifier.</param>
+        /// <param name="spanId"> A span identifier </param>
+        /// <param name="traceFlags">Contain details about the trace.</param>
+        /// <param name="traceState"> Carries system-specific configuration data.</param>
+        public ActivityContext(ActivityTraceId traceId, ActivitySpanId spanId, ActivityTraceFlags traceFlags, string? traceState = null)
+        {
+            // We don't allow creating context with invalid span or trace Ids.
+            if (spanId.ToHexString() == "0000000000000000" || traceId.ToHexString() == "00000000000000000000000000000000")
```

Comment:
I was trying to apply the OT specs as it is explicit about that. I am going to change the conditions to check the default instead of the contents but we need to be conscious we are not 100% conforming to the specs.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md

```
TraceId A valid trace identifier is a 16-byte array with at least one non-zero byte.
SpanId A valid span identifier is an 8-byte array with at least one non-zero byte.
```